pkg/sign/signer.go

```diff
@@ -149,7 +149,15 @@ func Bundle(content Content, keypair Keypair, opts BundleOptions) (*protobundle.
 			return nil, err
 		}
 
-		artifactOpts := verify.WithArtifact(bytes.NewReader(content.PreAuthEncoding()))
+		// Generally, you should provide an artifact when verifying.
+		//
+		// However, we just signed the DSSE object trusting the user has
+		// referenced the artifact(s) they intended.
+		artifactOpts := verify.WithoutArtifactUnsafe()
+		if bundle.GetMessageSignature() != nil {
```

Comment:
Why is this conditional necessary, and we only skip artifact verification?

Reply:
sigstore-go requires that some artifact reference is provided when verifying a message signature: https://github.com/sigstore/sigstore-go/blob/6f7e99d5908575d5faaddb5a688f83430ba989b1/pkg/verify/signature.go#L48

Reply:
Oh yea, that makes sense, because you need an artifact for a signature, but not for a DSSE since that's just checking subject claims